Auth0Tests/ClaimsValidatorsSpec.swift

```diff
@@ -0,0 +1,483 @@
+                    let expectedError = IDTokenIatValidator.ValidationError.missingIat
+                    let result = iatValidator.validate(jwt)
+                    
+                    expect(result).to(matchError(expectedError))
+                    expect(result?.errorDescription).to(equal(expectedError.errorDescription))
+                }
+            }
+            
+        }
+        
+        describe("nonce validation") {
+            
+            var nonceValidator: IDTokenNonceValidator!
+            let expectedNonce = "a1b2c3d4e5"
+            
+            beforeEach {
+                nonceValidator = IDTokenNonceValidator(nonce: expectedNonce)
+            }
+            
+            context("missing nonce") {
+                it("should return nil if nonce is present") {
+                    let jwt = generateJWT(nonce: expectedNonce)
+                    
+                    expect(nonceValidator.validate(jwt)).to(beNil())
+                }
+                
+                it("should return an error if nonce is missing") {
+                    let jwt = generateJWT(nonce: nil)
+                    let expectedError = IDTokenNonceValidator.ValidationError.missingNonce
+                    let result = nonceValidator.validate(jwt)
+                    
+                    expect(result).to(matchError(expectedError))
+                    expect(result?.errorDescription).to(equal(expectedError.errorDescription))
+                }
+            }
+            
+            context("mismatched nonce") {
+                it("should return nil if nonce matches the request nonce") {
+                    let jwt = generateJWT(nonce: expectedNonce)
+                    
+                    expect(nonceValidator.validate(jwt)).to(beNil())
+                }
+                
+                it("should return an error if nonce does not match the request nonce") {
+                    let nonce = "abc123"
+                    let jwt = generateJWT(nonce: nonce)
+                    let expectedError = IDTokenNonceValidator.ValidationError.mismatchedNonce(actual: nonce,
+                                                                                              expected: expectedNonce)
+                    let result = nonceValidator.validate(jwt)
+                    
+                    expect(result).to(matchError(expectedError))
+                    expect(result?.errorDescription).to(equal(expectedError.errorDescription))
+                }
+            }
+            
+        }
+        
+        describe("azp validation") {
+            
+            var azpValidator: IDTokenAzpValidator!
+            let expectedAzp = self.clientId
+            
+            beforeEach {
+                azpValidator = IDTokenAzpValidator(audience: expectedAzp)
+            }
+            
+            context("missing azp") {
+                it("should return nil if azp is present") {
```

Comment:
No need to, because this logic is not in the validator (nor should it be). It's in the top-level api function, and will be tested as a black box with integration tests.